### ISSUE TITLE
Rancher chart post-delete hook fails on hardened cluster

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#32878]
**What kind of request is this:**
bug

**Steps to reproduce:**

- Install Rancher on CIS 1.5 compliant cluster RKE1
  -  Details on pre-requirements [here](https://rancher.com/docs/rancher/v2.x/en/security/rancher-2.4/hardening-2.4/).
- Uninstall the Rancher chart `helm uninstall rancher -n cattle-system`

**Result:**

```
Events:
  Type     Reason        Age                From            Message
  ----     ------        ----               ----            -------
  Warning  FailedCreate  18s (x3 over 48s)  job-controller  Error creating: pods "rancher-post-delete-" is forbidden: PodSecurityPolicy: unable to admit pod: []
```

**Other details that may be helpful:**

**Environment information**
- Rancher version: `2.6 master-head (05/25/2021)` _f90b36e_
- Installation option: HA RKE1

**Comment by izaac** [comment]
Rancher version `2.6 master-head (06/06/2021)` _b0c0eb4_

This is failing now with:


```
k describe job rancher-post-delete -n cattle-system
Name:           rancher-post-delete
Namespace:      cattle-system
Selector:       controller-uid=65e6336a-20c8-4984-a894-657184783487
Labels:         app=rancher
                chart=rancher-2.5.8
                heritage=Helm
                release=rancher
Annotations:    helm.sh/hook: post-delete
                helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
                helm.sh/hook-weight: 3
Parallelism:    1
Completions:    1
Pods Statuses:  0 Running / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app=rancher
                    chart=rancher-2.5.8
                    controller-uid=65e6336a-20c8-4984-a894-657184783487
                    heritage=Helm
                    job-name=rancher-post-delete
                    release=rancher
  Service Account:  rancher-post-delete
  Containers:
   rancher-post-delete:
    Image:      rancher/shell:v0.1.6
    Port:       <none>
    Host Port:  <none>
    Command:
      /scripts/post-delete-hook.sh
    Environment:
      NAMESPACES:          fleet-system cattle-system rancher-operator-system
      TIMEOUT:             120
      IGNORETIMEOUTERROR:  false
    Mounts:
      /scripts from config-volume (rw)
  Volumes:
   config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      rancher-post-delete
    Optional:  false
Events:
  Type     Reason        Age                  From            Message
  ----     ------        ----                 ----            -------
  Warning  FailedCreate  46s (x5 over 3m16s)  job-controller  Error creating: pods "rancher-post-delete-" is forbidden: PodSecurityPolicy: unable to admit pod: [spec.containers[0].securityContext.runAsUser: Invalid value: 0: running with the root UID is forbidden]
```

**Comment by izaac** [comment]
Cluster has these pre-requirements configured/applied: https://rancher.com/docs/rancher/v2.x/en/security/rancher-2.5/1.6-hardening-2.5/

**Comment by jiaqiluo** [comment]
@izaac Sorry that I add the wrong label to the issue. We can not validate the fix until we have v2.6.0 rc

**Comment by deniseschannon** [comment]
Available to test in v2.6.0-rc1

**Comment by izaac** [comment]
Rancher version: `v2.6.0-rc1`

The post delete seems to be working now on a CIS install but there's a separate issue as fleet seems not getting cleared from the install.

Opened separate issue here: https://github.com/rancher/rancher/issues/33986
Original post-delete hook fix:  https://github.com/rancher/rancher/issues/30924
